pallets/pools/src/lib.rs

```diff
@@ -140,27 +160,56 @@ pub struct EpochDetails<BalanceRatio> {
 
 #[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, Default, TypeInfo)]
 pub struct EpochExecutionTranche<Balance, BalanceRatio> {
-	value: Balance,
+	supply: Balance,
 	price: BalanceRatio,
 	invest: Balance,
 	redeem: Balance,
+	min_risk_buffer: Perquintill,
+	seniority: Seniority,
 }
 
-/// The outstanding collections for an account
-#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, Default, TypeInfo)]
-pub struct OutstandingCollections<Balance> {
-	pub payout_currency_amount: Balance,
-	pub payout_token_amount: Balance,
-	pub remaining_invest_currency: Balance,
-	pub remaining_redeem_token: Balance,
+impl<Balance, BalanceRatio> TrancheWeigher for EpochExecutionTranche<Balance, BalanceRatio>
+where
+	Balance: From<u128>,
+{
+	type Weight = (Balance, Balance);
+	type External = NumTranches;
+
+	fn calculate_weight(&self, input: Self::External) -> Self::Weight {
```

Comment:
could `input` have a more semantic name?

Reply:
It basically represent external input. Maybe external or something?

Reply:
It's `n_tranches` in the other impl, Should it be the same here?